src/components/TodoList.tsx

```diff
@@ -0,0 +1,201 @@
+import cn from 'classnames';
+import { useContext, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { addTodo, deleteTodo } from '../api/todos';
+import { SetErrorContext } from '../utils/setErrorContext';
+
+interface Props {
+  todos: Todo[] | null,
+  filteringMode: string,
+  userId: number,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[] | null>>,
+  todosToBeDeleted: Todo['id'][] | null,
+  setTodosToBeDeleted: React.Dispatch<React.SetStateAction<number[] | null>>,
+}
+
+let filteredTodos: Todo[] | null = [];
+
+export const TodoList: React.FC<Props> = ({
+  todos, filteringMode, userId, setTodos, todosToBeDeleted, setTodosToBeDeleted,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [processing, setProcessing] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  if (filteringMode !== 'all' && todos !== null) {
```

Comment:
lets use enums for 'all', 'completed' etc
```suggestion
  if (filteringMode !== 'all' && todos) {
```

Reply:
> lets use enums for 'all', 'completed' etc